### PR TITLE
Add kong 1.0.x support

### PR DESCRIPTION
**Pull request by illyaMs**
This PR adds kong 1.0.x support (and 0.15 as well, since it's a pre-release for 1.0.0 with most of the breaking changes already on board).

Changes include:
- new schemas for 1.0.x and 0.15
- updated service & route contracts support
- new plugin contract support
- ability to run tests on win machines (via WSL)
- fix for null apikey header
- some minor ui fixes

Fixed the tests as well, at least it's a green street for 1.0.x, 0.15 and 0.14 versions.
(Did my best to keep backward compatibility with elder versions, but that's not guaranteed).

Note: certificate tests are disabled when running Kong >= 0.15. Seems like kong has an improved cert/key params validation, which I wasn't able to overcome, unfortunately. Hope someone else will get more luck with that part.

### CODE REVIEW

**Review by sebastiandelaroche — APPROVED** [review]

File: README.md
Comment (PGBI) [inline README.md]: @illyaMs First of all, thanks for this great work. And sorry for taking so time to respond... 

I believe Kong is following semver, so it should be safe to assume that the next version of Kong dashboard that you produced will be compatible with Kong < 2.0.0 and not just < 1.1.0.

Also, I was considering making it a major Kong-dashboard release (4.0.0) and dropping support for Kong < 1.0. The codebase, is becoming more and more difficult to read, especially tests. Now that Kong should be stable, I think it's time to stop supporting the 0.x versions. What do you think?

Comment (illyaMs) [reply README.md]: @PGBI , no problem, nice to see you're still here 👍 

Agree, I will change the `1.1.0` hardcodes to `2.0.0` everywhere and commit right now.

Speaking of dropping the pre-1.0.x support - I can only completely agree. Was thinking of it each time while adding one more hard-coded version check in tests, but obviously wasn't able to make this decision on my own :)

Maybe we can leave the `>= 0.9, <2.0.0` support for v3.6.0 (to deliver existing work done with minimal effort and unblock the community to use the dashboard right away)? And agree to have a next 4.0.0 release with a major cleanup (so me or anyone else who's capable to commit on that part in the nearest future may have their time to work on that)?

**Comment by guoyang1982** [comment]
hi,when can i use it?

**Comment by asifpy** [comment]
@PGBI When this will get merged?

File: README.md
Comment (PGBI) [reply README.md]: sounds good to me. Will do the merge and release this weekend. Thanks for your patience.

**Comment by nileshb30** [comment]
Hi,
 How to resolve Kong dashboard 3.5.0 version doesn't support Kong v0.15 and higher.I am trying installing another version of kong but it can not install.How to resolve this.I install the kong 1.1.1 latest version and kong-dashboard 3.6.0.But the dashboard is not showing all the tab it only showing the HOME,PLUGINS and SERVICES tab.How can i get remaining tab in the kong dashboard.

**Comment by illyaMs** [comment]
> Hi,
> How to resolve Kong dashboard 3.5.0 version doesn't support Kong v0.15 and higher.I am trying installing another version of kong but it can not install.How to resolve this.I install the kong 1.1.1 latest version and kong-dashboard 3.6.0.But the dashboard is not showing all the tab it only showing the HOME,PLUGINS and SERVICES tab.How can i get remaining tab in the kong dashboard.

Actually `kong 1.x.x` should work fine with `kong-dashboard 3.6.0`.
Are you sure you're using proper versions? It's strange that you see `Services` tab and don't see `Routes` one.
Can you also please check if you have the same dashboard problem with `kong 1.0.x` version?

**Comment by nileshb30** [comment]
Hi i tried the kong 1.0.0 and dashboard 3.6.0 ,3.4.0,3.5.0 but the dashboard are not supported for the kong 1.0.0 please check below logs in my terminal and suggest me where i am going wrong.And please share correct version of kong an dashboard supported.And also check the image attached of kong dashboard which is not showing API tab.
kongDashboard


pc@nilesh-PC MINGW64 /c/Program Files/Docker Toolbox
$ docker run --rm -p 8085:8080 pgbi/kong-dashboard:v3.6.0 start --kong-url http://192.168.99.100:8001
**Unable to find image 'pgbi/kong-dashboard:v3.6.0' locally**
**C:\Program Files\Docker Toolbox\docker.exe: Error response from daemon: manifest for pgbi/kong-dashboard:
v3.6.0 not found.**
See 'C:\Program Files\Docker Toolbox\docker.exe run --help'.

pc@nilesh-PC MINGW64 /c/Program Files/Docker Toolbox
$ docker run --rm -p 8085:8080 pgbi/kong-dashboard:v3.4.0 start --kong-url http://192.168.99.100:8001
Unable to find image 'pgbi/kong-dashboard:v3.4.0' locally
v3.4.0: Pulling from pgbi/kong-dashboard
88286f41530e: Pull complete
d0e8a23136b3: Pull complete
5ad5b12a980e: Pull complete
ac005de72214: Pull complete
6a2ad6a5de96: Pull complete
Digest: sha256:f9b199a6148cbd1f018e8074825b5f0679216f60b2ea340cec9fdd2589c56918
Status: Downloaded newer image for pgbi/kong-dashboard:v3.4.0
C:\Program Files\Docker Toolbox\docker.exe: Error response from daemon: driver failed programming externa
l connectivity on endpoint sleepy_ardinghelli (13338ccb9a2ae7f064d3a1f0aecb17e31532f7522cd07da86919940f16
f7d959): Bind for 0.0.0.0:8085 failed: port is already allocated.

pc@nilesh-PC MINGW64 /c/Program Files/Docker Toolbox
$ docker run --rm -p 8086:8080 pgbi/kong-dashboard:v3.4.0 start --kong-url http://192.168.99.100:8001
Connecting to Kong on http://192.168.99.100:8001 ...
**This version of Kong dashboard doesn't support Kong v0.14 and higher.**

pc@nilesh-PC MINGW64 /c/Program Files/Docker Toolbox
$ docker run --rm -p 8086:8080 pgbi/kong-dashboard:v3.5.0 start --kong-url http://192.168.99.100:8001
Unable to find image 'pgbi/kong-dashboard:v3.5.0' locally
v3.5.0: Pulling from pgbi/kong-dashboard
8e3ba11ec2a2: Already exists
e883c863a46d: Already exists
f84102466f54: Already exists
030d2eaba018: Pull complete
770ccbebddd0: Pull complete
Digest: sha256:2cb728f4d5ec42db3f8b8cf7d54f268ce34183e96329eab21f5dec3654b48220
Status: Downloaded newer image for pgbi/kong-dashboard:v3.5.0
Connecting to Kong on http://192.168.99.100:8001 ...
**This version of Kong dashboard doesn't support Kong v0.15 and higher.**

**Comment by illyaMs** [comment]
Speaking of `v3.4.0` and `v3.4.5` - they are not supposed to support Kong v.1.x.x., so your console logs tells you the pure truth :)

Speaking of `v.3.4.6` - seems like the tag is missing in docker hub, you can use the `v3` tag instead (see all the image tags [here](https://hub.docker.com/r/pgbi/kong-dashboard/tags)).

@PGBI could you also please add the `v.3.4.6` docker image tag too? I know `v3` pointing to latest should be ok, but seems like it may confuse someone expecting the full version tag.

File: README.md
Comment (capcom923) [reply README.md]: I installed kong1.0.3 and this dashboard 3.6.0. The left menu are OK. But the comsumer's oauth2 plugin still not work well. The return url could not be set.

Comment (illyaMs) [reply README.md]: > I installed kong1.0.3 and this dashboard 3.6.0. The left menu are OK. But the comsumer's oauth2 plugin still not work well. The return url could not be set.

Some of the plugins were tested for compatibility, but I don't remember exactly if the oauth2 one was on the list. I think there were some Unit Tests covering it, so actually I think it should work.

Could you provide more details (ideally in separate github issue, linked to this MR/issue)?

**Pull request merged**
